### DISCUSSION TITLE
Why is highlighting different if search repeated with s<Cr>

### DISCUSSION THREAD

**Discussion by ctstacey** [ggandor/leap.nvim#102] [General]
Why is the highlighting different when a search is repeated? (I.e. using s<Cr>)

**Comment by ctstacey** [comment]
To clarify, I'm talking about the highlighting when a search is repeated using `s<CR>`

![image](https://user-images.githubusercontent.com/51082452/216868785-18674d32-c756-42d0-8d23-22ceeb7eaa52.png)

**Comment by ggandor** [comment]
You don't have to type the pattern when repeating, so there's no need to shift the label, we can display it covering the whole match, in classic Sneak-style. I think of this as a feature, a visual indication of the current "mode" (wide beacons mean "these are the final targets, no pattern input is expected, choose now").

**Reply by ctstacey** [reply]
Are there any other things you can do in this 'mode'?

**Reply by ggandor** [reply]
No, everything works the same way, you just skip right to the last phase, as if you've entered the two characters manually.

**Reply by ctstacey** [reply]
Hmmm, ok.

My initial thought was that if nothing were functionally different, the highlights should not be different.

But the more I ponder, the more sense it makes to signal the difference in 'mode'.
It becomes a visual confirmation, a re-enforcement, before the leap is made.
Which becomes more useful as you chain more complicated actions together (eg. leap-spooky).

Nice.
Thanks for responding and **nice work with the plugins**!!!
